src/dso_api/dynamic_api/serializers.py

```diff
@@ -286,6 +263,21 @@ def _raise_better_embed_error(cls, field_name, prefix=""):
                     f" for embedding as it's a summary of a huge listing."
                 ) from None
 
+    def get_embedded_objects_by_id(
```

Comment:
I really like this refactor! Much cleaner now and totally in the spirit of DRF.

Reply:
yeah moving this logic to the serializer avoided this `id_fetcher` mess and made overriding much easier. Took some time to figure that one out! :-D